include/zephyr/posix/pthread.h

```diff
@@ -480,6 +480,18 @@ int pthread_atfork(void (*prepare)(void), void (*parent)(void), void (*child)(vo
 int pthread_getconcurrency(void);
 int pthread_setconcurrency(int new_level);
 
+void __z_pthread_cleanup_push(void *cleanup[3], void (*routine)(void *arg), void *arg);
+void __z_pthread_cleanup_pop(int execute);
+
+#define pthread_cleanup_push(_rtn, _arg)                                                           \
+	do /* enforce '{'-like behaviour */ {                                                      \
+		void *_z_pthread_cleanup[3];                                                       \
```

Comment:
Will this cause compiler warnings if you nest these?

Reply:
I don't believe so - there is a test and I don't see any warnings when building it.